cscs-checks/system/slurm/slurm.py

```diff
@@ -198,26 +200,20 @@ def set_memory_limit(self):
 
 @rfm.simple_test
 class MemoryOverconsumptionMpiCheck(SlurmCompiledBaseCheck):
-    def __init__(self):
-        super().__init__()
-        self.maintainers = ['JG']
-        self.valid_systems += ['eiger:mc', 'pilatus:mc']
-        self.time_limit = '5m'
-        self.sourcepath = 'eatmemory_mpi.c'
-        self.tags.add('mem')
-        self.executable_opts = ['100%']
-        self.sanity_patterns = sn.assert_found(r'(oom-kill)|(Killed)',
-                                               self.stderr)
-        # {{{ perf
-        regex = (r'^Eating \d+ MB\/mpi \*\d+mpi = -\d+ MB memory from \/proc\/'
-                 r'meminfo: total: \d+ GB, free: \d+ GB, avail: \d+ GB, using:'
-                 r' (\d+) GB')
-        self.perf_patterns = {
-            'max_cn_memory': sn.getattr(self, 'reference_meminfo'),
-            'max_allocated_memory': sn.max(
-                sn.extractall(regex, self.stdout, 1, int)
-            ),
-        }
+    maintainers = ['JG']
+    valid_systems += ['eiger:mc', 'pilatus:mc']
+    time_limit = '5m'
+    sourcepath = 'eatmemory_mpi.c'
+    tags.add('mem')
+    executable_opts = ['100%']
+
+    @sanity_function
+    def assert_found_oom(self):
+        return sn.assert_found(r'(oom-kill)|(Killed)',
+                               self.stderr)
+
+    @run_before('performance')
+    def set_perf_patterns(self):
```

Comment:
```suggestion
    def set_reference(self):
```

Reply:
The `reference_meminfo()` needs not to be any or a deferrable property. It is accessed in a `performance_function` or a pre-performance hook, so you can simply make it a function that returns the reference value that you want.